### PR TITLE
chore: optimize scala performance

### PR DESCRIPTION
**Pull request by He-Pin**
IIRC, Scala 2.13 does the trick where it transform the for  to while loop, have not checked the Scala 3.6.x
docs: https://scala-cli.virtuslab.org/docs/reference/directives

### CODE REVIEW

**Comment by He-Pin** [comment]
@WojciechMazur Does Scala 3 transform the `for` into a `while loop`?
It's slower than Java currently

**Comment by He-Pin** [comment]
@bddicken If naive fib is required I can paritally revert it.

File: fibonacci/scala/code.scala [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
+    r += fibonacci(i, 0, 1)
+    i += 1
   println(r)
 
+@tailrec
+def fibonacci(n: Int, a: Int, b: Int): Int =
+  if (n > 0)
+    fibonacci(n - 1, b, a + b)
+  else
+    a
```
Comment (axman6) [inline fibonacci/scala/code.scala:22]: As far as I understand it, the point of this benchmark is to test the naive non-tail recursive implementation. Every language would be faster if it used the linear time algorithm.

Comment (He-Pin) [reply fibonacci/scala/code.scala:22]: seems true, let's me revert it

**Comment by He-Pin** [comment]
Currently , Range.foreach is not optimized to while loop
https://github.com/scala/scala3/issues/8880
https://github.com/scala/bug/issues/1338

**Comment by He-Pin** [comment]
@bddicken Please merge this , thanks.

**Comment by bddicken** [comment]
Thanks for submitting this 👍

**Pull request merged**